client/matching/client.go

```diff
@@ -192,18 +192,30 @@ func (c *clientImpl) QueryWorkflow(
 	ctx context.Context,
 	request *types.MatchingQueryWorkflowRequest,
 	opts ...yarpc.CallOption,
-) (*types.QueryWorkflowResponse, error) {
+) (*types.MatchingQueryWorkflowResponse, error) {
 	partition := c.loadBalancer.PickReadPartition(
 		persistence.TaskListTypeDecision,
 		request,
 		"",
 	)
+	originalTaskListName := request.TaskList.GetName()
 	request.TaskList.Name = partition
 	peer, err := c.peerResolver.FromTaskList(request.TaskList.GetName())
 	if err != nil {
 		return nil, err
 	}
-	return c.client.QueryWorkflow(ctx, request, append(opts, yarpc.WithShardKey(peer))...)
+	resp, err := c.client.QueryWorkflow(ctx, request, append(opts, yarpc.WithShardKey(peer))...)
+	if err != nil {
+		return nil, err
+	}
+	request.TaskList.Name = originalTaskListName
+	c.provider.UpdatePartitionConfig(
+		request.GetDomainUUID(),
+		*request.TaskList,
+		persistence.TaskListTypeDecision,
+		resp.PartitionConfig,
```

Comment:
As this rolls out the PartitionConfig will be nil, what will be the behavior of that? If we roll out the change to history before updating matching to return a value will that result in every query task clearing the cache?

Reply:
It's noop for nil